### DISCUSSION TITLE
Are GraphQL Argument Directives supported?

### DISCUSSION THREAD

**Discussion by jonnydgreen** [MichalLytek/type-graphql#1043] [Q&A]
I'm looking to create the following schema using `type-graphql`:

```graphql
type Query {
  message(id: ID @constraint(...etc)): Message
}
```

I had an idea that the directive could be defined with the following class:

```ts
@ArgsType()
export class MessageArguments {
  @Directive('@constraint(...etc)')
  @Field(() => ID, { description: "The ID of the message.", nullable: true })
  id?: string;
}
```

But am unable to get it to work (note, the astNode on the argument is undefined). I found the following which suggests that directives on arguments isn't yet supported?
 - https://typegraphql.com/docs/directives.html#declaring-in-schema
 - https://github.com/MichalLytek/type-graphql/pull/369#issuecomment-510277254

But I wasn't 100% so wanted to double check! :)

If not, I wondered if this is planned to be supported and if not, whether we could discuss any alternative approaches I have also considered?

**Comment by MichalLytek** [comment]
Of course it is, see the examples:
https://github.com/MichalLytek/type-graphql/blob/master/examples/apollo-federation/products/product.ts#L3

**Reply by MichalLytek** [reply]
Ah you mean on arguments? Read about the limitations:
> Currently, you can use the directives only on object types, input types and their fields or fields resolvers, as well as queries, mutations and subscriptions.

**Reply by jonnydgreen** [reply]
Cool thanks for the confirmation. If you think it's a good idea, I'm very happy to prepare a PR to add Directive support on GraphQL field arguments? :) If I were to do so, would this be feasible to release in v1 or would you prefer this to ship with the next release?

**Reply by MichalLytek** [reply]
v1 is released, we can ship such improvements in minor release 1.2

**Reply by jonnydgreen** [reply]
Cool, thanks - I'll start looking at what is needed to add this support :)
